ros2-library/src/main/java/us/ihmc/ros2/ROS2NodeBuilder.java

```diff
@@ -0,0 +1,271 @@
+               LogTools.warn("Using a default ROS Domain ID of {}", domainId);
+            }
+         }
+
+         profile.domainId(domainId);
+      }
+
+      // Set up transports
+      {
+         profile.useBuiltinTransports(false);
+
+         if (useSharedMemory)
+            profile.addSharedMemoryTransport();
+
+         profile.addUDPv4Transport(addressRestriction);
+
+         if (transportSpecial != null)
+         {
+            switch (transportSpecial)
+            {
+               case SHARED_MEMORY_ONLY ->
+               {
+                  profile.useOnlySharedMemoryTransport();
+               }
+               case LOOPBACK_ADDRESS_ONLY ->
+               {
+                  InetAddress loopbackAddress = InetAddress.getLoopbackAddress();
+
+                  addressRestriction = new InetAddress[] {loopbackAddress};
+               }
+               case UDPV4_ONLY ->
+               {
+                  profile.useOnlyUDPv4Transport(addressRestriction);
+               }
+            }
+         }
+      }
+
+      // Print the current transports
+
+      return profile;
+   }
+
+   public ROS2Node build(String name)
+   {
+      return new ROS2Node(name, namespace, buildProfile());
+   }
+
+   public RealtimeROS2Node buildRealtime(String name)
+   {
+      return buildRealtime(name, new PeriodicNonRealtimeThreadSchedulerFactory());
+   }
+
+   public RealtimeROS2Node buildRealtime(String name, PeriodicThreadSchedulerFactory threadFactory)
+   {
+      return new RealtimeROS2Node(name, namespace, buildProfile(), threadFactory);
+   }
+
+   protected static boolean domainIDValid(int domainID)
+   {
+      return domainID >= 0 && domainID <= 232;
+   }
+
+   /**
+    * Find the ROS Domain ID from either:
+    *    - IHMCNetworkParameters.ini
+    *    - System property
+    *    - Environment variable
+    * prioritized in that order
+    */
+   protected static int findDomainID(boolean parseNetworkParametersConfig, boolean parseProperties, boolean parseEnvironment)
+   {
+      int domainID = UNSET_DOMAIN_ID;
+
+      // Find domain ID from IHMCNetworkParameters.ini
+      if (parseNetworkParametersConfig)
+      {
+         File networkParametersFile = new File(System.getProperty("user.home"), ".ihmc/IHMCNetworkParameters.ini");
+         Properties properties = new Properties();
+         try (FileInputStream inputStream = new FileInputStream(networkParametersFile))
+         {
+            properties.load(inputStream);
+
+            String domainIDProperty = properties.getProperty("RTPSDomainID");
+
+            if (domainIDProperty != null)
+            {
+               try
+               {
+                  domainID = Integer.parseInt(domainIDProperty);
+               }
+               catch (NumberFormatException e)
+               {
+                  // Ignore
```

Comment:
Maybe we should print something when there's a format exception? It means that the parameters file has an issue with it, right?